### PR TITLE
Govendor

### PR DESCRIPTION
**Pull request by mwhooker**
fresh golang:1.6 docker build works with these changes

closes #3947 #3948 #3949 #3955

### CODE REVIEW

**Comment by mwhooker** [comment]
need to test the following

- [ ] go 1.4 build still works
- [ ] make deps works
- [ ] `make vendor` works

**Comment by mwhooker** [comment]
for the go 1.6 build I ran ```
go get github.com/mitchellh/packer
git co govendor -t origin/govendor
sh ./scripts/build.sh
```

and it built just fine

**Comment by rickard-von-essen** [comment]
Great :+1:

**Comment by mwhooker** [comment]
Was able to remove a lot of unused dependencies.
also tested with go1.5.

want to get a few +1s then will merge

File: Makefile [hunk 1]
```diff
@@ -29,11 +29,8 @@ package:
 deps:
 	go get github.com/mitchellh/gox
 	go get golang.org/x/tools/cmd/stringer
-	@go version | grep 1.4 ; if [ $$? -eq 0 ]; then \
-		echo "Installing godep and restoring dependencies"; \
-		go get github.com/tools/godep; \
-		godep restore; \
-	fi
+	go get -u github.com/kardianos/govendor
```
Comment (cbednarski) [inline Makefile]: This invokes a network call and git sync each time you run make. I recommend no `-u` or wrap in `if` block.

File: Makefile [hunk 2]
```diff
@@ -73,13 +70,12 @@ testrace: deps ## Test for race conditions
...
 	go get -u golang.org/x/tools/cmd/stringer
-	@echo "INFO: Packer deps are managed by godep. See CONTRIBUTING.md"
+	@echo "INFO: Packer deps are managed by govendor. See CONTRIBUTING.md"
 
 # This is used to add new dependencies to packer. If you are submitting a PR
 # that includes new dependencies you will need to run this.
 vendor: ## Add new dependencies.
-	godep restore
-	godep save
+	@echo "INFO: Packer deps are managed by govendor. See CONTRIBUTING.md"
```
Comment (cbednarski) [inline Makefile]: If we are not running anything here I would just delete the target completely.

File: builder/azure/arm/template_factory.go [hunk 3]
```diff
@@ -8,6 +8,8 @@ import (
 
 	"github.com/mitchellh/packer/builder/azure/common/constants"
 	"github.com/mitchellh/packer/builder/azure/common/template"
+
+	_ "github.com/approvals/go-approval-tests"
```
Comment (mwhooker) [inline builder/azure/arm/template_factory.go]: hmm, I suppose we could manually add it to vendor.json (or do nothing as it's in there already) and take out this line.

I don't see any govendor utilities for pulling in tests

**Comment by rickard-von-essen** [comment]
Awesome!

**Pull request merged**